### PR TITLE
Import loader.webp without process.env

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

Use Vite static asset handling to import `loader.webp`.

https://vitejs.dev/guide/assets.html

Note:

- because we explicitely set chunk name in the build config, the keep its name during build process (as currently) - i.e. no timestamp is added to the filename as displayed in the documentation

- not sure it would make contribution easy but we could transform `dapps.json` to a `dapps.ts` and use the same pattern if we want to

### CODE REVIEW

File: src/frontend/env.d.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+/// <reference types="vite/client" />
+declare module "envConfig";
```
Comment (nmattia) [inline src/frontend/env.d.ts:2]: What is this for?

File: jest-transform.cjs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ *
+ * e.g.
+ * import loaderUrl from "../../assets/loader.webp";
+ */
+module.exports = {
+  process(sourceText, sourcePath, options) {
+    return {
+      code: `module.exports = ${JSON.stringify(path.basename(sourcePath))};`,
+    };
+  },
```
Comment (nmattia) [inline jest-transform.cjs:14]: should this somehow be limited by file extension?

File: src/frontend/env.d.ts (hunk 1 above)
Comment (peterpeterparker) [reply src/frontend/env.d.ts:2]: > TypeScript, by default, does not recognize static asset imports as valid modules. To fix this, include [vite/client](https://vitejs.dev/guide/features.html#client-types).

https://vitejs.dev/guide/assets.html

File: jest-transform.cjs (hunk 2 above)
Comment (peterpeterparker) [reply jest-transform.cjs:14]: It's done by the transformer pattern of the configuration

```
"\\.(svg|png|webp)$": "<rootDir>/jest-transform.cjs",
```

File: src/frontend/env.d.ts (hunk 1 above)
Comment (nmattia) [reply src/frontend/env.d.ts:2]: Can you add a note about this in the file?

Comment (peterpeterparker) [reply src/frontend/env.d.ts:5]: Added a link to the doc

**Review by nmattia — APPROVED** [review]
Thanks!

**Pull request merged**